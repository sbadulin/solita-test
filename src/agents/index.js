export { default as productsAgent } from "./products";
