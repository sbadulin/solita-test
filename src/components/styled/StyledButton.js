import styled from "styled-components";

import { COLORS } from "../../constants";

export const StyledButton = styled.button`
  width: ${props => props.width || "190px"};
  padding: 12px 24px;
  font-size: 18px;
  font-family: "Playfair Display", serif;
  font-weight: 600;
  border-color: ${COLORS.MAIN};
  font-style: normal;
  line-height: 18px;
  letter-spacing: 0.9px;
  color: ${COLORS.MAIN};
  cursor: pointer;
  background-color: #fff;

  @media (min-width: 600px) {
    &:not(:last-child) {
      margin-right: 10px;
    }
  }
  @media (max-width: 600px) {
    margin-top: 10px;
  }

  &:hover {
    background-color: #eee;
  }
  &:active {
    background-color: #ddd;
  }
  &:focus {
    outline: 0;
  }
`;
