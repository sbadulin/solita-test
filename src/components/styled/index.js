export { StyledRow } from "./StyledRow";
export { StyledColumn } from "./StyledColumn";
export { StyledItemCard } from "./StyledItemCard";
export { StyledItemList } from "./StyledItemList";
export { StyledNavWrapper } from "./StyledNavWrapper";
export { StyledHeader } from "./StyledHeader";
export { StyledLink } from "./StyledLink";
export { StyledButton } from "./StyledButton";
export { StyledItemCardPrice } from "./StyledItemCardPrice";
export { StyledProductImage } from "./StyledProductImage";
export { StyledPageWrapper } from "./StyledPageWrapper";
export { StyledLogo } from "./StyledLogo";
export { StyledProductInfo } from "./StyledProductInfo";
