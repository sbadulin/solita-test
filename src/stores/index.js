export { default as productsStore } from "./products";
export { default as cartStore } from "./cart";
