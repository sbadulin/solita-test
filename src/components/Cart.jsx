import React from "react";
import { inject, observer } from "mobx-react";
import { Link } from "@reach/router";
import {
  StyledButton,
  StyledItemCardPrice,
  StyledItemList,
  StyledRow,
  StyledColumn,
  StyledPageWrapper
} from "./styled";
import { ItemCard } from "./ItemCard";
import { CURRENCY_SYMBOL } from "../constants";
import { InfoMessage } from "./styled/Typography";

import { cartInfo } from "./utils/cartInfo";

const Cart = inject(stores => ({
  addToCart: stores.cartStore.addToCart,
  removeFromCart: stores.cartStore.removeFromCart,
  cartItemIds: stores.cartStore.cartItemIds,
  products: stores.productsStore.registry
}))(
  observer(({ products, addToCart, removeFromCart, cartItemIds }) => {
    const { items, total } = cartInfo(cartItemIds, products);
    const handleAddToCart = id => () => addToCart(id);
    const handleRemoveFromCart = id => () => removeFromCart(id);
    return (
      <StyledPageWrapper>
        {cartItemIds.length > 0 ? (
          <StyledItemList>
            {items.map(item => (
              <ItemCard {...item}>
                <StyledItemCardPrice>{`${item.count} Ã— ${
                  item.price
                } ${CURRENCY_SYMBOL}`}</StyledItemCardPrice>
                <StyledRow>
                  <StyledButton
                    width={80}
                    onClick={handleRemoveFromCart(item.id)}
                  >
                    {"-".toUpperCase()}
                  </StyledButton>
                  <StyledButton width={80} onClick={handleAddToCart(item.id)}>
                    {"+".toUpperCase()}
                  </StyledButton>
                </StyledRow>
              </ItemCard>
            ))}
            <InfoMessage>
              Total: <span className="totalPrice">{total}</span>
              &#8239;{CURRENCY_SYMBOL}
            </InfoMessage>
          </StyledItemList>
        ) : (
          <StyledColumn align="center">
            <InfoMessage>Cart is empty</InfoMessage>
            <Link to="/">
              <StyledButton width={280}>
                {"Go to products list".toUpperCase()}
              </StyledButton>
            </Link>
          </StyledColumn>
        )}
      </StyledPageWrapper>
    );
  })
);

export default Cart;
