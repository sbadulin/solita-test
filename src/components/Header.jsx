import React from "react";
import { inject } from "mobx-react";

import { Link } from "@reach/router";

import { HeaderBody } from "./styled/Typography";
import {
  StyledHeader,
  StyledRow,
  StyledLink,
  StyledNavWrapper,
  StyledLogo
} from "./styled";
import { COLORS } from "../constants";

import ShoppingCartIcon from "../assets/shopping-cart.svg";

const Header = inject(stores => ({ cartSize: stores.cartStore.cartSize }))(
  ({ cartSize }) => (
    <StyledHeader>
      <nav>
        <StyledNavWrapper>
          <StyledRow>
            <Link to="/">
              <StyledLogo />
            </Link>
            <HeaderBody>
              The most definitive shape store in the world
            </HeaderBody>
          </StyledRow>
          <StyledRow>
            <StyledLink
              to="cart"
              getProps={({ isCurrent }) => ({
                style: { fill: isCurrent ? COLORS.ACCENT : COLORS.MAIN }
              })}
            >
              <StyledRow>
                <HeaderBody style={{ marginRight: "6px" }}>
                  {cartSize ? <strong>{cartSize}</strong> : "No"} items in cart
                </HeaderBody>
                <ShoppingCartIcon width={52} height={52} />
              </StyledRow>
            </StyledLink>
          </StyledRow>
        </StyledNavWrapper>
      </nav>
    </StyledHeader>
  )
);

export default Header;
