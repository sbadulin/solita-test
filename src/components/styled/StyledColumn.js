import styled from "styled-components";

export const StyledColumn = styled.div`
  display: flex;
  flex-direction: column;
  align-items: ${props => (props.align ? props.align : "flex-start")};
  ${props => props.grow && "flex-grow: 1;"}

  &:not(:last-child) {
    margin-right: 40px;

    @media (max-width: 600px) {
      margin-right: 0;
    }
  }

  @media (max-width: 600px) {
    align-items: center;
  }
`;
