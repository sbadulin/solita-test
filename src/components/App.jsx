import React from "react";
import { Router } from "@reach/router";
import { Normalize } from "styled-normalize";
import { Provider } from "mobx-react";

import { productsStore, cartStore } from "../stores";
import ItemList from "./ItemList";
import Cart from "./Cart";
import Header from "./Header";

const stores = { productsStore, cartStore };

const App = () => (
  <Provider {...stores}>
    <>
      <Normalize />
      <Header />
      <Router>
        <ItemList path="/" />
        <Cart path="cart" />
      </Router>
    </>
  </Provider>
);

export default App;
