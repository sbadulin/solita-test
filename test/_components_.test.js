import React from "react";
import { render } from "react-testing-library";
import "jest-styled-components";
import { Provider } from "mobx-react";

import { productsStore, cartStore } from "../src/stores";
import App from "../src/components/App";
import Header from "../src/components/Header";
import Cart from "../src/components/Cart";
import { StyledButton } from "../src/components/styled/StyledButton";

test("App renders", () => {
  const stores = { productsStore, cartStore };
  const { container } = render(
    <Provider {...stores}>
      <App />
    </Provider>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("Header renders", () => {
  const stores = { productsStore, cartStore };
  const { container } = render(
    <Provider {...stores}>
      <Header />
    </Provider>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("Cart renders", () => {
  const stores = { productsStore, cartStore };
  const { container } = render(<Provider {...stores}><Cart /></Provider>);
  expect(container.firstChild).toMatchSnapshot();
});

test("Button renders", () => {
  const { container } = render(<StyledButton />);
  expect(container.firstChild).toMatchSnapshot();
});
