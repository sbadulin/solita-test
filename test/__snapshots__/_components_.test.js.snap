// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 {
  position: relative;
  padding: 20px 30px 60px 35px;
  margin-bottom: 30px;
  background: linear-gradient(180deg,#ececec 0%,#e4e4e4 100%);
}

.c0:after {
  background: linear-gradient(-45deg,#ffffff 25px,transparent 0), linear-gradient(45deg,#ffffff 25px,transparent 0) repeat-x left bottom;
  background-size: 50px 50px;
  content: " ";
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 50px;
}

.c5 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:visited {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover {
  color: #EF4A8D;
}

.c3 {
  margin-right: 20px;
  fill: #000;
  width: 127px;
  height: 64px;
}

.c4 {
  display: inline-block;
  font-family: "Playfair Display",sans-serif;
  font-size: 16px;
  font-weight: 400;
  font-style: italic;
  font-stretch: normal;
  line-height: 2.25;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  padding: 8px;
  margin: 0;
}

.c4 strong {
  font-weight: 600;
}

@media (max-width:600px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c3 {
    margin-right: 0;
  }
}

@media (max-width:600px) {
  .c4 {
    font-size: 20px;
    text-align: center;
    line-height: 1.5;
  }
}

<header
  class="c0"
>
  <nav>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          aria-current="page"
          href="/"
        >
          <svg
            class="c3"
          />
        </a>
        <p
          class="c4"
        >
          The most definitive shape store in the world
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          class="c5"
          href="/cart"
          style="fill: #000;"
        >
          <div
            class="c2"
          >
            <p
              class="c4"
              style="margin-right: 6px;"
            >
              No
               items in cart
            </p>
            <svg
              height="52"
              width="52"
            />
          </div>
        </a>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Button renders 1`] = `
.c0 {
  width: 190px;
  padding: 12px 24px;
  font-size: 18px;
  font-family: "Playfair Display",serif;
  font-weight: 600;
  border-color: #000;
  font-style: normal;
  line-height: 18px;
  -webkit-letter-spacing: 0.9px;
  -moz-letter-spacing: 0.9px;
  -ms-letter-spacing: 0.9px;
  letter-spacing: 0.9px;
  color: #000;
  cursor: pointer;
  background-color: #fff;
}

.c0:hover {
  background-color: #eee;
}

.c0:active {
  background-color: #ddd;
}

.c0:focus {
  outline: 0;
}

@media (min-width:600px) {
  .c0:not(:last-child) {
    margin-right: 10px;
  }
}

@media (max-width:600px) {
  .c0 {
    margin-top: 10px;
  }
}

<button
  class="c0"
/>
`;

exports[`Cart renders 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:not(:last-child) {
  margin-right: 40px;
}

.c3 {
  width: 280;
  padding: 12px 24px;
  font-size: 18px;
  font-family: "Playfair Display",serif;
  font-weight: 600;
  border-color: #000;
  font-style: normal;
  line-height: 18px;
  -webkit-letter-spacing: 0.9px;
  -moz-letter-spacing: 0.9px;
  -ms-letter-spacing: 0.9px;
  letter-spacing: 0.9px;
  color: #000;
  cursor: pointer;
  background-color: #fff;
}

.c3:hover {
  background-color: #eee;
}

.c3:active {
  background-color: #ddd;
}

.c3:focus {
  outline: 0;
}

.c0 {
  padding: 20px;
}

.c2 {
  font-family: "Playfair Display",sans-serif;
  font-style: italic;
  font-weight: bold;
  line-height: 60px;
  font-size: 36px;
  -webkit-letter-spacing: 0.36px;
  -moz-letter-spacing: 0.36px;
  -ms-letter-spacing: 0.36px;
  letter-spacing: 0.36px;
  color: #000;
  margin: 0 0 10px;
  text-align: center;
}

.c2 span.totalPrice {
  font-size: 1.6em;
}

@media (max-width:600px) {
  .c1:not(:last-child) {
    margin-right: 0;
  }
}

@media (max-width:600px) {
  .c1 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:600px) {
  .c3:not(:last-child) {
    margin-right: 10px;
  }
}

@media (max-width:600px) {
  .c3 {
    margin-top: 10px;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      Cart is empty
    </p>
    <a
      aria-current="page"
      href="/"
    >
      <button
        class="c3"
        width="280"
      >
        GO TO PRODUCTS LIST
      </button>
    </a>
  </div>
</main>
`;

exports[`Header renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 {
  position: relative;
  padding: 20px 30px 60px 35px;
  margin-bottom: 30px;
  background: linear-gradient(180deg,#ececec 0%,#e4e4e4 100%);
}

.c0:after {
  background: linear-gradient(-45deg,#ffffff 25px,transparent 0), linear-gradient(45deg,#ffffff 25px,transparent 0) repeat-x left bottom;
  background-size: 50px 50px;
  content: " ";
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 50px;
}

.c5 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:visited {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover {
  color: #EF4A8D;
}

.c3 {
  margin-right: 20px;
  fill: #000;
  width: 127px;
  height: 64px;
}

.c4 {
  display: inline-block;
  font-family: "Playfair Display",sans-serif;
  font-size: 16px;
  font-weight: 400;
  font-style: italic;
  font-stretch: normal;
  line-height: 2.25;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  padding: 8px;
  margin: 0;
}

.c4 strong {
  font-weight: 600;
}

@media (max-width:600px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c3 {
    margin-right: 0;
  }
}

@media (max-width:600px) {
  .c4 {
    font-size: 20px;
    text-align: center;
    line-height: 1.5;
  }
}

<header
  class="c0"
>
  <nav>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          aria-current="page"
          href="/"
        >
          <svg
            class="c3"
          />
        </a>
        <p
          class="c4"
        >
          The most definitive shape store in the world
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          class="c5"
          href="/cart"
          style="fill: #000;"
        >
          <div
            class="c2"
          >
            <p
              class="c4"
              style="margin-right: 6px;"
            >
              No
               items in cart
            </p>
            <svg
              height="52"
              width="52"
            />
          </div>
        </a>
      </div>
    </div>
  </nav>
</header>
`;
